Pre-QA Dapp Stabilization

rdig opened the pull request:
## Description

This PR fixes the rough outstanding issue that will prevent us from launching. It handles mostly problems we saw while initially testing the Dapp on QA.

The issues fixed here are meant as a stop-gap, rather than a polished "final" version, as most of them can be improved later down the line

**Changes**

- [x] Pinata.cloud IPFS client
- [x] Setup the IPFS clients with fallback to one another
- [x] Setup the IPFS clients get / add actions on a timeout of 10 secs
- [x] Added polling, w/ timeout, to the getCode method used by the providers _by @area_
- [x] Refactored `ColonyHome` loading and 404 logic
- [x] `SubscribedColoniesList` show a preloader until data is ready
- [x] Optimized the way we fetch the user's subscribed to colonies _(Note: we have **hard** technical limit at 100 colonies)_
- [x] `ColonyHome` show a loading when changing the colonies from the sidebar _(subscriptions list)_
- [x] Added `/funding` route _(`ColonyFunding` component)_ loading state
- [x] Removed old, unused routes: admin, task
- [x] Properly fetch role events for pre v5 colonies, so that upgraded colonies have correct user permissions _(this is a `colonyJS` fix actually -- released version `3.0.0-alpha.19`)_ 
- [x] Prevent payment action from being made, if the `OneTxPayment` is not installed
- [x] Disable user actions if the Metamask wallet network is "incorrect"
- [x] Added `/members` route _(`ColonyMembers` component)_ loading state
- [x] Added a local client resolver to fetch the metacolony (if available)
- [x] Added `/landing` route _(`LadingPage` component)_ loading state
- [x] Refactored the way we upload / get user and colonies avatars from IPFS (pinata) in order to support more than 250 KB file sizes
- [x] Optimized actions sagas transaction orchestration wait times _(due to IPFS lag)_

Note: this can, and should be tested directly on QA: http://qa.colony.io/landing

**Screenshots, _where relevant_:**

Prevent payment action, when the OneTxPayment extension is not installed:
![Screenshot from 2021-01-30 00-09-07](https://user-images.githubusercontent.com/1193222/106333033-8caea300-6290-11eb-8376-b0da4b4df57e.png)
![Screenshot from 2021-01-30 00-14-39](https://user-images.githubusercontent.com/1193222/106333035-8d473980-6290-11eb-832c-2196b5a00a4a.png)


Among the above, this PR also fixes, in passing:

Resolves DEV-162
Resolves DEV-166
Resolves DEV-167
Resolves DEV-178

alicjakujawa reviewed (changes requested):
Small comments/suggestions. Will test it now

chinins reviewed (commented):
Seems good to me other than some of Alicja's comments.  I have tested only locally and probably not everything. Will keep testing tomorrow.

rdig commented on src/modules/dashboard/components/ManageDomainsDialog/ManageDomainsDialog.tsx:
```diff
@@ -83,7 +83,8 @@ const ManageDomainsDialog = ({
         permissionRequired: <FormattedMessage {...MSG.domainPermissionsList} />,
       },
-      onClick: () => callStep(nextStep),
+      // onClick: () => callStep(nextStep),
```
It's just temporary. _(that's why it's only commented out)_

I got annoyed about people reporting as problems, issues that were clearly specified as _not yet merged_

So I just disabled them. We'll enable them back one by one as we merge them in

rdig commented on src/data/resolvers/metacolony.ts:
```diff
@@ -0,0 +1,62 @@
+        return null;
+      } catch (error) {
+        console.error(error);
+        return null;
```
Does it matter?

That's the point of all those `null` returns if something goes wrong along the way have a consistent value in the components to act upon (yes, in this case `null`)

That means that I don't really care if it's a chain error, or subgraph error. If I get null in the component when calling this resolver, then I know to not show a link to the metacolony

rdig commented on src/modules/users/components/HookedUserAvatar/HookedUserAvatar.ts:
```diff
@@ -28,11 +29,16 @@ export default withHooks<
+  try {
+    avatarObject = JSON.parse(avatar);
+  } catch (error) {
+    // silent error
```
Why would I want to show the error here? 

1. We have two fallbacks here, if no url is available, we try to fetch, if we can't fetch it we default to a blockie
2. All errors that can show up in here would have already been caught by IPFS

I see not point in showing a similar error, to ones coming from IPFS for the exact same problem.

rdig commented on src/modules/dashboard/components/ColonyFundingMenu/ColonyFundingMenu.tsx:
```diff
@@ -80,30 +76,45 @@ const ColonyFundingMenu = ({
+
+  const isSupportedColonyVersion =
+    parseInt(version, 10) >= ColonyVersion.CeruleanLightweightSpaceship;
+  const isNetworkAllowed = !!ALLOWED_NETWORKS[networkId || 1];
```
Yes, which is ethereum mainnet: 

https://github.com/JoinColony/colonyDapp/blob/f53547841df4800f4a2d93d1b001027779a4efd7/src/modules/constants.ts#L98-L103

Since `ETHEREUM_NETWORK.chainId` is `1`

rdig commented on src/data/resolvers/metacolony.ts:
```diff
@@ -0,0 +1,62 @@
+               * Converting this to string basically converts the error object into
+               * a string form
+               */
+              address: metaColonyAddress.toLowerCase(),
```
I spent too long trying to remember what I was trying to say here. Turns out I just copy-pasted this whole section and were it doesn't make sense :facepalm: 

It was from here: https://github.com/JoinColony/colonyDapp/blob/f53547841df4800f4a2d93d1b001027779a4efd7/src/data/resolvers/colony.ts#L314-L340

I'll remove the comment since it's confusing and wrong.

rdig commented:
@JoinColony/frontend This is now rebased, fixed (based on Alicja's last review), and awaiting your set of eyes again :eyes: 

Hopefully, we can merge this on Monday to get this monkey off our backs

ArmandoGraterol reviewed (approved):
Looks good to me! I tested various actions available and they seem to be working fine

alicjakujawa reviewed (approved):
Seems to working great 🔥  but never ever again so huge PRs 🤣

rdig commented on src/modules/dashboard/components/ColonyFundingMenu/ColonyFundingMenu.tsx:
```diff
@@ -80,30 +76,58 @@ const ColonyFundingMenu = ({
+            !isSupportedColonyVersion ||
+            !isNetworkAllowed ||
+            !hasRegisteredProfile ||
+            !isDeploymentFinished
```
I agree. That's why I opened this issue to address this: https://linear.app/colony/issue/DEV-172/create-an-access-system-available-from-the-context

chinins reviewed (approved):
All good from my side as well!
